modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaSttpClientCodegen.java

```diff
@@ -212,21 +213,25 @@ public String escapeReservedWord(String name) {
             try {
                 @SuppressWarnings("unchecked")
                 Map<String, ArrayList<CodegenOperation>> opsMap = (Map<String, ArrayList<CodegenOperation>>) objs.get("operations");
-                HashSet<Integer> unknownCodes = new HashSet<Integer>();
+                HashSet<Integer> unknownCodes = new HashSet<>();
                 for (CodegenOperation operation : opsMap.get("operation")) {
+                    boolean hasOnlyDefaultResponse = true;
                     for (CodegenResponse response : operation.responses) {
-                        if ("default".equals(response.code)) {
+                        if (response.isDefault) {
                             continue;
                         }
                         try {
                             int code = Integer.parseInt(response.code);
+                            hasOnlyDefaultResponse = false;
                             if (code >= 600) {
                                 unknownCodes.add(code);
                             }
+                            response.vendorExtensions.put("x-error-code", response.is4xx || response.is5xx);
```

Comment:
what does `vendorExtensions` provide?

Reply:
It's used to determine whether response should be `Left` or `Right` in template generator. See `ignoredResponseReturnType.mustache`. Is there any better way to achieve this?

Reply:
I see; no idea tbh never used mustache